sat/cli/jobstat/main.py

```diff
@@ -42,10 +42,10 @@ def do_jobstat(args):
     """
     state_checker_client = StateCheckerClient(SATSession())
     try:
-        if args.all is True:
-            application_data = state_checker_client.get_all()
+        application_data = state_checker_client.get_all()
+        if application_data['jobstat'] == None:
```

Comment:
I am a little confused by the most recent change on this line, but I think you probably meant to do this as `is not None` rather than `== None`, otherwise you are only creating the report if the response looks like this:
```
{"jobstat": null}
```
(i.e. it contained a `'jobstat'` key and the value was `NoneType`).

With this change, I also don't think it's going to be correctly handling the case when the response is missing a `jobstat` key, the command will fail with an uncaught `KeyError` coming from this line.

Assuming you did mean to say `if application_data['jobstat'] is not None`, then If the response does contain a `jobstat` key but returns an empty list, I think you will get an uncaught `IndexError` on line 48 trying to get the 0th element of an empty list.

I think it would be good to also account for the case when the response contains the `jobstat` key but a `null` value (i.e. that case I show at the top of this comment), once again I think that would result in an unhandled exception coming from line 48.

Just in general I think it would still be cleaner to be returning just the value pointed to by the `"jobstat"` key - this version makes the `except KeyError` on line 55 of `state_checker.py` no longer necessary - and I don't think we are correctly handling that case now.

For me I think I would still prefer something along these lines, though there might be cases that I am not thinking of: https://gist.github.hpe.com/eli-kamin/aa8f60f88dd465507f6410ee568e98ba

Reply:
Are we attempting to handle the case when the API returns both valid jobstat results as well as an error message? It seems like maybe you are trying to do that, not sure.

Reply:
Our response always has jobstat e.g. 
`{'machine': 'slice', 'errorcode': 2, 'errormessage': 'Unable to query Elasticsearch', 'timereported': '2022-09-20T08:26:02', 'jobstat': []}`
We have machine just in case for multi tenancy.  A non-zero errorcode indicates failure so we use that instead of suggested checks.

Reply:
Response structure is always the same from the server, just values change.